#!/usr/bin/env python

PACKAGE='jsk_perception'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("ALL", bool_t, 0, "All attributes ", True)
gen.add("DEFAULT", bool_t, 0, "Default attributes, BoundingBox, Confidence, Landmarks, Pose, and Quality", False)

# # https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/Rekognition/TFaceDetail.html
# gen.add("AgeRange", bool_t, 0, "The estimated age range, in years, for the face. Low represents the lowest estimated age and High represents the highest estimated age.", False)
# gen.add("Beard", bool_t, 0, "Indicates whether or not the face has a beard, and the confidence level in the determination.", False)
# gen.add("BoundingBox", bool_t, 0, "Bounding box of the face. Default attribute.", False)
# gen.add("Confidence", bool_t, 0, "Confidence level that the bounding box contains a face (and not a different object such as a tree). Default attribute.", False)
# gen.add("Emotions", bool_t, 0, "The emotions that appear to be expressed on the face, and the confidence level in the determination. The API is only making a determination of the physical appearance of a person's face. It is not a determination of the person's internal emotional state and should not be used in such a way. For example, a person pretending to have a sad face might not be sad emotionally.", False)
# gen.add("Eyeglasses", bool_t, 0, "Indicates whether or not the face is wearing eye glasses, and the confidence level in the determination.", False)
# gen.add("EyesOpen", bool_t, 0, "Indicates whether or not the eyes on the face are open, and the confidence level in the determination.", False)
# gen.add("Gender", bool_t, 0, "The predicted gender of a detected face.", False)
# gen.add("Landmarks", bool_t, 0, "Indicates the location of landmarks on the face. Default attribute.", False)
# gen.add("MouthOpen", bool_t, 0, "Indicates whether or not the mouth on the face is open, and the confidence level in the determination.", False)
# gen.add("Mustache", bool_t, 0, "Indicates whether or not the face has a mustache, and the confidence level in the determination.", False)
# gen.add("Pose", bool_t, 0, "Indicates the pose of the face as determined by its pitch, roll, and yaw. Default attribute.", False)
# gen.add("Quality", bool_t, 0, "Identifies image brightness and sharpness. Default attribute.", False)
# gen.add("Smile", bool_t, 0, "Indicates whether or not the face is smiling, and the confidence level in the determination.", False)
# gen.add("Sunglasses", bool_t, 0, "Indicates whether or not the face is wearing sunglasses, and the confidence level in the determination.", False)

exit(gen.generate(PACKAGE, "jsk_perception", "AWSDetectFaces"))
